### PR TITLE
BUG: avoid ResourceWarnings in the amrex frontend

### LINKED ISSUE #4890: BUG: ResourceWarnings for unclosed files in boxlib frontend
### Bug report

**Bug summary**

`open()` is used several times in the boxlib frontend without closing the files, triggering a `ResourceWarning` when the files go out of scope. Originally noticed by @neutrinoceros in https://github.com/yt-project/yt/issues/4845#issuecomment-2029735171.

**Code for reproduction**

Standalone reproducer using the `ParticleCavity` dataset:
```python
import warnings
import yt

# enable all warnings
warnings.simplefilter("default")
ds = yt.load("plt.Cavity00010")
```

**Actual outcome**

```
/home/eric/build/yt/yt/frontends/boxlib/data_structures.py:701: ResourceWarning: unclosed file <_io.TextIOWrapper name='/scratch/yt-data/plt.Cavity00010/inputs' mode='r' encoding='UTF-8'>
  lines = [line.lower() for line in open(cparam_filepath).readlines()]
ResourceWarning: Enable tracemalloc to get the object allocation traceback
/home/eric/build/yt/yt/frontends/boxlib/data_structures.py:728: ResourceWarning: unclosed file <_io.TextIOWrapper name='/scratch/yt-data/plt.Cavity00010/Header' mode='r' encoding='UTF-8'>
  self._parse_header_file()
ResourceWarning: Enable tracemalloc to get the object allocation traceback
yt : [INFO     ] 2024-04-30 14:59:50,921 Parameters: current_time              = 0.017408426036316
yt : [INFO     ] 2024-04-30 14:59:50,921 Parameters: domain_dimensions         = [16 16  1]
yt : [INFO     ] 2024-04-30 14:59:50,921 Parameters: domain_left_edge          = [0. 0. 0.]
yt : [INFO     ] 2024-04-30 14:59:50,922 Parameters: domain_right_edge         = [1. 1. 1.]
```

**Expected outcome**

No warnings.

**Version Information**
  * yt version: 4.4.dev0, cd4f1d69cd3f (current git head)

### PR DESCRIPTION
**Pull request by yut23**
## PR Summary

Use context managers where the line length allows, and manually close all other files that are opened outside a `with` statement.

Fixes #4890.

### CODE REVIEW

**Review by neutrinoceros — APPROVED** [review]
This seems uncontroversial but I never got why we were leaving handles opened in the first place, so I'd like to leave this open for a little while in case someone objects.

**Review by matthewturk — DISMISSED** [review]
Good change.  My suggestions are optional.  Thanks for catching this -- I'm not sure why we left handles open!

File: yt/frontends/boxlib/data_structures.py [hunk 1]
```diff
@@ -734,7 +740,8 @@ def _parse_parameter_file(self):
     def _parse_cparams(self):
         if self.cparam_filename is None:
             return
-        for line in (line.split("#")[0].strip() for line in open(self.cparam_filename)):
+        param_file = open(self.cparam_filename)
```
[the remark this answers is not in the document]
Comment (yut23) [reply yt/frontends/boxlib/data_structures.py]: Using context managers everywhere would be best, but I didn't want to go too crazy with the indentation. I just updated this function, along with a couple of one-liners elsewhere.

[the remark this answers is not in the document]
Comment (matthewturk) [reply yt/frontends/boxlib/data_structures.py]: Sounds good to me!

**Review by neutrinoceros — APPROVED** [review]

**Pull request merged**